docs/design/third-party-libraries.md

```diff
@@ -0,0 +1,175 @@
+That data can be anything. Any keys starting with either `$` or `_` will be ignored and not
+passed to the client. It could be used for builds or scaffolding information.
+
+During installation, there's a step to prompt the user for data. The schema contains a prompt
+text and a default value. The default value type is used to convert the string entered by the
+user. The key used in `appPrompt` is the key saved in appData.
+
+Example:
+
+```typescript
+{ // ...
+  "angular-cli": {
+    "appPrompt": {
+      "number": {
+        "prompt": "Please enter a number:",
+        "defaultValue": 0
+      },
+      "url": {
+        "prompt": "URL of your website:",
+        "defaultValue": "${homepage}"
+      },
+    }
+  }
+}
+```
+
+The default value is necessary as a `quiet` mode is enforced, using default values to fill
+in the data without user interaction. The special strings `${...}` can be used to replace
+with special values in the default string values. The values are taken from the `package.json`.
+
+We use a declarative style to enforce sensible defaults and make sure developers think about
+this. *In the case where developers want more complex interaction, they can use the
+install/uninstall hooks to prompt users.* But 3rd party libraries developers need to be aware
+that those hooks might not prompt users (no STDIN) and throw an error.
+
+# Providers
+Adding Angular providers to the app should be seamless. The build process will create a
+`providers.js` from all the providers contained in all the dependencies. The User can blacklist
+providers it doesn't want.
+
+# Dependencies
+The install process should be recursive to allow for dependencies. However, the process should
+fail if there is a version difference between two libraries depending on the same library. This
+differs significantly from, say, `npm` where a `node_modules` can exist within another module.
+For the sake of simplicity, we do not support this case and simply throw.
+
+# Index.html
+
+We do not touch the `index.html` file during the installation task. The default page should
+link to a SystemJS configuration file that is auto generated by the CLI and not user
+configurable (see SystemJS below). If the user install a third party library, like jQuery, and
+wants to use it, they have to import it using `import * as $ from 'vendor/jquery'`.
+
+# SystemJS
+It is important that SystemJS works without any modifications by the user. It is also important
+to leave the liberty to the user to change the SystemJS configuration so that it fits their needs.
+
+We will not use SystemJS bundles in development. This is for better debugging, future proofing
+(when moving the build system) and better CDN support, as many of the loaded files will end up
+being pulled from a CDN in production. During the `ng build` process for production, the
+SystemJS configuration script will be rebuilt to fetch from the CDN.
```

Comment:
It feels a bit superfluous to have a strong commitment to SystemJS (for the CDN functionality) and still keep the brocolli build step. 

Perhaps we could move to SystemJS TS browser compilation for dev builds, and have the production build step simply compile TS to spare SystemJS from doing it.